File: Foyer/sol.java
import java.util.Arrays;
import java.util.Scanner;

public class Sol {

    public static int sol(String[] items, String price) {
        int itemsCount = 0;
        int totalPrice = Integer.parseInt(price.substring(1));

        // Convert and sort prices
        int[] prices = new int[items.length];
        for (int i = 0; i < items.length; i++) {
            prices[i] = Integer.parseInt(items[i].substring(1));
        }
        Arrays.sort(prices);

        // Calculate max items
        for (int item : prices) {
            if (item <= totalPrice) {
                itemsCount++;
                totalPrice -= item;
            } else {
                break;
            }
        }

        return itemsCount == 0 ? "Insufficient cash!" : itemsCount;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String[] itemsList = scanner.nextLine().split(" ");
        String totalPrice = scanner.next();

        System.out.println(sol(itemsList, totalPrice));
        scanner.close();
    }
}

File: Foyer/sol.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int compare(const void *a, const void *b) {
    return (*(int *)a - *(int *)b);
}

int maxItems(char *list[], char *price) {
    int itemsCount = 0;
    int totalPrice = atoi(price + 1);

    // Convert and sort prices
    int numItems = 0;
    while (list[numItems] != NULL) {
        numItems++;
    }
    int *items = (int *)malloc(numItems * sizeof(int));
    for (int i = 0; i < numItems; i++) {
        items[i] = atoi(list[i] + 1);
    }
    qsort(items, numItems, sizeof(int), compare);

    // Calculate max items
    for (int j = 0; j < numItems; j++) {
        if (items[j] <= totalPrice) {
            itemsCount++;
            totalPrice -= items[j];
        } else {
            break;
        }
    }

    free(items);

    return itemsCount == 0 ? "Insufficient cash!" : itemsCount;
}

int main() {
    char *itemsList[100];
    char totalPrice[10];

    char input[1000];
    fgets(input, sizeof(input), stdin);
    int i = 0;
    char *token = strtok(input, " \n");
    while (token != NULL) {
        itemsList[i] = strdup(token);
        token = strtok(NULL, " \n");
        i++;
    }
    itemsList[i] = NULL;

    scanf("%s", totalPrice);

    printf("%d", maxItems(itemsList, totalPrice));

    return 0;
}
